### ISSUE TITLE
A9: Model Evaluation & Deployment Pipeline

### ISSUE THREAD

**Issue by amolk99** [othmanreem/Data-intensive-systems#38]
## Objective
Evaluate trained models, identify optimal configurations, and create a retrainable deployment pipeline.

### Model Evaluation
- [ ] Evaluate DL model predictions on test set
- [ ] Calculate regression metrics:
  - Mean Squared Error (MSE)
  - Mean Absolute Error (MAE)
  - R² Score
- [ ] Per-joint evaluation:
  - Assess prediction quality for each of 13 joints
  - Identify joints with highest/lowest prediction error
- [ ] Visualize actual vs predicted Z coordinates

### Architecture Comparison
- [ ] Compare Dense vs Conv1D vs LSTM/GRU performance
- [ ] Document trade-offs:
  - Training time
  - Inference speed
  - Prediction accuracy
  - Model complexity
- [ ] Create comparison plots and tables

### Optimal Setup Selection
- [ ] Rank all tested configurations by performance
- [ ] Select champion model based on:
  - Best MAE/MSE on validation
  - Reasonable training time
  - Good generalization (low overfitting)
- [ ] Document optimal hyperparameters:
  - Architecture type
  - Number of layers/nodes
  - Optimizer and learning rate
  - Batch size and epochs
- [ ] Justify selection with metrics

### Retraining Pipeline
- [ ] Create modular, retrainable model code:
  ```python
  class DepthPredictor:
      def __init__(self, config):
          self.model = self.build_model(config)
      
      def train(self, data, epochs, callbacks):
          ...
      
      def predict(self, xy_coordinates):
          ...
      
      def save(self, path):
          ...
      
      def load(self, path):
          ...
  ```
- [ ] Implement configuration file for easy retraining
- [ ] Create training script with CLI arguments
- [ ] Document retraining procedure

### Model Persistence
- [ ] Save best model weights: `A9/models/best_model.keras`
- [ ] Save model configuration: `A9/models/config.json`
- [ ] Implement model loading for inference
- [ ] Test model serialization/deserialization

### Prediction Usage
- [ ] Create inference pipeline:
  ```python
  # Usage example
  predictor = DepthPredictor.load('best_model.keras')
  xy_input = [[head_x, head_y, lshoulder_x, lshoulder_y, ...]]  # 26 features
  z_output = predictor.predict(xy_input)  # 13 z-coordinates
  ```
- [ ] Document input/output formats
- [ ] Create example usage script

## Acceptance Criteria
- All trained models evaluated with regression metrics
- Clear comparison between architecture types
- Optimal setup identified with justification
- Retrainable pipeline with easy configuration
- Model can be saved and loaded successfully
- Usage examples work end-to-end
- Dead ends documented (what didn't work and why)

## Dependencies
- Requires: Issue #36 (Data Loading & Architecture)
- Requires: Issue #37 (Model Training & Hyperparameter Tuning)

## Related Files
- Output: `A9/predictor.py`
- Output: `A9/models/best_model.keras`
- Output: `A9/models/config.json`
- Output: `A9/inference.py`
